service-base/src/main/java/org/eclipse/hono/service/credentials/BaseCredentialsService.java

```diff
@@ -130,6 +130,9 @@ public final void processCredentialsMessage(final Message<JsonObject> regMsg) {
         } catch (ClassCastException e) {
             log.debug("malformed request message");
             reply(regMsg, CredentialsResult.from(HTTP_BAD_REQUEST));
+        } catch (NullPointerException ne) {
```

Comment:
I agree and had similar thoughts already. Will improve it and later on change it for the registration service as well.

Ecplicit checks will improve the detailed log message as well which is IMHO appropriate for a well defined API (and would be too much possibly for parts deep down in the implementation classes).